src/api/firebase.ts

```diff
@@ -245,12 +245,15 @@ export async function addItem(
 	});
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+export async function updateItem(listPath: string, item: ListItem) {
+	const itemDocRef = doc(db, listPath, "items", item.id);
+
+	const updates: Partial<ListItem> = {};
+
+	updates.totalPurchases = item.totalPurchases + 1;
+	updates.dateLastPurchased = Timestamp.fromDate(new Date());
+
+	await updateDoc(itemDocRef, updates);
```

Comment:
Suggestion: We can maybe wrap this in a `try...catch` block to handle potential errors in case the `updateDoc` call fails?

Reply:
Good catch!

Reply:
Yes, great catch @zahrafalak !